src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -53,6 +53,30 @@ typedef struct gavl_tree {
 
 #define STACK_EMPTY(stack) (!stack##_sp)
 
+#if MPL_THREAD_PACKAGE_NAME != MPL_THREAD_PACKAGE_NONE
+
+#define GAVL_THREAD_CS_ENTER                                                              \
+    do {                                                                                  \
+        if (gavl_tree_iptr->need_thread_safety) {                                         \
+            MPL_thread_mutex_lock(&gavl_tree_iptr->lock, &mpl_err, MPL_THREAD_PRIO_HIGH); \
+            if (mpl_err != MPL_SUCCESS)                                                   \
+                return mpl_err;                                                           \
+        }                                                                                 \
+    } while (0)
+
+#define GAVL_THREAD_CS_EXIT                                           \
+    do {                                                              \
+        if (gavl_tree_iptr->need_thread_safety)                       \
+            MPL_thread_mutex_unlock(&gavl_tree_iptr->lock, &mpl_err); \
+    } while (0)
```

Comment:
Please don't add more macros that assume local variables (e.g., `gavl_tree_iptr`).  That's bad coding practice (although admittedly we do that in some places in MPICH).

Reply:
Fixed.